astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -2,13 +2,16 @@
 # This module implements the Arithmetic mixin to the NDData class.
 
 import warnings
+from collections.abc import Buffer
```

Comment:
Bah, looks like this was added in Python 3.12, which is why there are tests failing now :slightly_frowning_face:.

Reply:
Actually, passing a string or buffer to `np.result_type()` in that block of code would not work, because it tries to interpret the str/buffer value as the name of a data type, so we _should_ exclude all of the above types, so that my extra logic doesn't break some exotic usage.